solana/modules/wormhole-anchor-sdk/src/token_bridge/accounts.rs

```diff
@@ -164,25 +166,98 @@ impl TransferWithPayload {
     }
 }
 
-impl AnchorDeserialize for TransferWithPayload {
-    fn deserialize(buf: &mut &[u8]) -> io::Result<Self> {
+impl Writeable for TransferWithPayload {
+    fn write<W>(&self, writer: &mut W) -> io::Result<()>
+    where
+        W: io::Write,
+    {
+        let Self { meta, payload } = self;
+
+        meta.write(writer)?;
+        payload.write(writer)?;
+
+        Ok(())
+    }
+
+    fn written_size(&self) -> usize {
+        let Self { meta, payload } = self;
+
+        meta.written_size() + payload.written_size()
+    }
+}
+
+impl Readable for TransferWithPayload {
+    const SIZE: Option<usize> = None;
+
+    fn read<R>(reader: &mut R) -> io::Result<Self>
+    where
+        Self: Sized,
+        R: io::Read,
+    {
         Ok(TransferWithPayload {
-            meta: TransferWithMeta::deserialize(&mut &buf[..133])?,
-            payload: buf[133..].to_vec(),
+            meta: TransferHeader::read(reader)?,
+            payload: WriteableBytes::read(reader)?.into(),
         })
     }
 }
 
-#[derive(Default, AnchorSerialize, Clone, PartialEq, Eq)]
+impl AnchorSerialize for TransferWithPayload {
+    fn serialize<W: Write>(&self, writer: &mut W) -> io::Result<()> {
+        self.write(writer)
+    }
+}
+
+impl AnchorDeserialize for TransferWithPayload {
+    fn deserialize_reader<R: io::Read>(reader: &mut R) -> io::Result<Self> {
+        Readable::read(reader)
+    }
+}
```

Comment:
Would be great if we could get rid of this boilerplate somehow. Given that Anchor* are foreign traits, the best we can do is probably a macro_rules! ?

Reply:
I think that it's not worth it to add the complexity of a macro, only not to repeat 3 times the same code. Would it be 20 times that I would have written it.

Reply:
It's only 3 times here, but this will be common boiler-plate for anyone who uses wormhole-io together with Anchor.